tests/nodes/atomShake.cpp

```diff
@@ -0,0 +1,104 @@
+
+        auto atomicMCNode = std::make_unique<AtomicMCNode>(static_cast<Graph *>(loop_));
+        ASSERT_TRUE(atomicMCNode->setOption<Number>("ShakesPerAtom", 10));
+        atomicMCNode_ = atomicMCNode.get();
+        ASSERT_TRUE(loop_->addNode(std::move(atomicMCNode), "AtomicMC"));
+
+        // Water species
+        auto species = createWater(&root_);
+
+        // Water configuration
+        configurationNode_ = dynamic_cast<ConfigurationNode *>(root_.createNode("Configuration", "Bulk"));
+        ASSERT_TRUE(configurationNode_);
+        ASSERT_TRUE(configurationNode_->setOption<Number>("Temperature", 300.0));
+
+        insertNode_ = dynamic_cast<InsertNode *>(root_.createNode("Insert", "Insert"));
+        ASSERT_TRUE(insertNode_);
+
+        auto population = insertNode_->findInput("Population");
+        population->set<Number>(1);
+
+        auto density = insertNode_->findInput("Density");
+        density->set<Number>(9.99999642e-05);
+
+        // Set density units
+        ASSERT_TRUE(insertNode_->setOption<Units::DensityUnits>("DensityUnits", Units::DensityUnits::AtomsPerAngstromUnits));
+
+        // Set connections
+        EXPECT_TRUE(root_.addEdge({"Water", "Species", "Insert", "Species"}));
+        EXPECT_TRUE(root_.addEdge({"Bulk", "Configuration", "Insert", "Configuration"}));
+        EXPECT_TRUE(root_.addEdge({"Insert", "Configuration", "Loop", "Configuration"}));
+        EXPECT_TRUE(loop_->addEdge({"Inputs", "Configuration", "AtomicMC", "Configuration"}));
+        // TODO: Output parameter name cannot be the same as loopback (although loopback param CAN match input).
+        // May need to enforce this.
+        EXPECT_TRUE(loop_->addEdge({"AtomicMC", "Configuration", "Outputs", "OutputConfiguration"}));
+        EXPECT_TRUE(loop_->addEdge({"AtomicMC", "Configuration", "LoopBacks", "Configuration"}));
+    }
+
+    protected:
+    // We need a CoreData and Dissolve definition to properly instantiate DissolveGraph at present.
+    CoreData coreData_;
+    Dissolve dissolve_;
+    DissolveGraph root_;
+    ConfigurationNode *configurationNode_{nullptr};
+    SpeciesNode *speciesNode_{nullptr};
+    AtomicMCNode *atomicMCNode_{nullptr};
+    InsertNode *insertNode_{nullptr};
+    LoopGraph *loop_{nullptr};
+};
+
+TEST_F(AtomShakeTest, Water)
+{
+    auto nLoops = 100;
+    createGraph(nLoops);
+    ASSERT_TRUE(loop_->run() == NodeConstants::ProcessResult::Success);
+
+    // Check the geometry of the molecule
+    auto cfg = configurationNode_->getOutputValue<Configuration *>("Configuration");
+    auto *box = cfg->box();
+    auto &atoms = cfg->atoms();
+    auto rMin01 = box->minimumDistance(atoms[0].r(), atoms[1].r());
+    auto rMin02 = box->minimumDistance(atoms[0].r(), atoms[2].r());
+    auto angle102 = box->angleInDegrees(atoms[1].r(), atoms[0].r(), atoms[2].r());
+    EXPECT_NEAR(rMin01, 1.012, 1.0e-4);
+    EXPECT_NEAR(rMin02, 1.012, 1.0e-4);
+    EXPECT_NEAR(angle102, 113.24, 1.7e-2);
```

Comment:
Here we should test for the default forcefield geometry as defined:

```suggestion
    EXPECT_NEAR(rMin01, 1.0, 1.0e-4);
    EXPECT_NEAR(rMin02, 1.0, 1.0e-4);
    EXPECT_NEAR(angle102, 113.24, 1.7e-3);
```